tests/app/test_model.py

```diff
@@ -657,3 +678,1616 @@ def test_serialize_service_filter_keys(
+    notification = create_notification(template=sample_template)
+    notify_db_session.commit()
+
+    serialized = notification.serialize()
+
+    assert serialized["id"] == notification.id
+    assert serialized["reference"] == notification.client_reference
+    if notification.notification_type == EMAIL_TYPE:
+        assert serialized["email_address"] == notification.to
+    else:
+        assert serialized["email_address"] is None
+    assert serialized["phone_number"] == notification.to if notification.notification_type == SMS_TYPE else None
+    assert serialized["type"] == notification.notification_type
+    assert serialized["status"] == notification.status
+    assert serialized["template"].get("id") == notification.template.id
+    assert serialized["body"] == notification.template.content
+    assert serialized["subject"] == notification.subject
+    assert serialized["created_at"]  # UTC string representation
+    assert serialized["created_by_name"] is None
+    assert serialized["sent_at"] is None
+    assert serialized["completed_at"] is None
+    assert serialized["scheduled_for"] is None
+    assert serialized["postage"] is None
+    assert serialized["one_click_unsubscribe_url"] is None
+
+
+def test_notification_serialization_only_returns_defined_fields(client, notify_db_session, sample_template):
+    notification = create_notification(template=sample_template)
+    notify_db_session.commit()
+
+    serialized = notification.serialize()
+
+    # Ensure only the defined fields are present in the serialized output
+    assert set(serialized.keys()) == {
+        "id",
+        "reference",
+        "email_address",
+        "phone_number",
+        "line_1",
+        "line_2",
+        "line_3",
+        "line_4",
+        "line_5",
+        "line_6",
+        "postcode",
+        "type",
+        "status",
+        "template",
+        "body",
+        "subject",
+        "created_at",
+        "created_by_name",
+        "sent_at",
+        "completed_at",
+        "scheduled_for",
+        "postage",
+        "one_click_unsubscribe_url",
+        "cost_in_pounds",
+        "cost_details",
+        "is_cost_data_ready",
+    }
+
+
+def test_notification_serialization_with_cost_data_returns_all_fields(client, notify_db_session, sample_template):
+    rate = Rate(
+        notification_type=SMS_TYPE,
+        rate=0.05,
+        valid_from=datetime(2024, 1, 1, tzinfo=UTC),
+    )
+    notify_db_session.add(rate)
+    notification = create_notification(template=sample_template)
+    notify_db_session.commit()
+
+    serialized = notification.serialize_with_cost_data()
+
+    assert serialized["id"] == notification.id
+    assert serialized["cost_details"] is not None
+    assert serialized["cost_in_pounds"] == 0.05
+    assert serialized["is_cost_data_ready"] is True
+
+
+def test_invited_organisation_user_serialization_returns_all_fields(
```

Comment:
If we want to check all fields, you could add an `assert serialized["permissions"] == `